src/funcs.rs

```diff
@@ -0,0 +1,237 @@
+        [
+            // numeric
+            FunctionContainer::new("abs", FunctionWrapper::Float1(f64::abs)),
+            FunctionContainer::new("acos", FunctionWrapper::Float1(f64::acos)),
+            FunctionContainer::new("asin", FunctionWrapper::Float1(f64::asin)),
+            FunctionContainer::new("atan", FunctionWrapper::Float1(f64::atan)),
+            FunctionContainer::new("atan2", FunctionWrapper::Float2(f64::atan2)),
+            FunctionContainer::new("cbrt", FunctionWrapper::Float1(f64::cbrt)),
+            FunctionContainer::new("ceil", FunctionWrapper::Float1(f64::ceil)),
+            FunctionContainer::new("cos", FunctionWrapper::Float1(f64::cos)),
+            FunctionContainer::new("cosh", FunctionWrapper::Float1(f64::cosh)),
+            FunctionContainer::new("exp", FunctionWrapper::Float1(f64::exp)),
+            FunctionContainer::new("expm1", FunctionWrapper::Float1(f64::exp_m1)),
+            FunctionContainer::new("floor", FunctionWrapper::Float1(f64::floor)),
+            FunctionContainer::new("hypot", FunctionWrapper::Float2(f64::hypot)),
+            FunctionContainer::new("log", FunctionWrapper::Float1(f64::ln)),
+            FunctionContainer::new("log10", FunctionWrapper::Float1(f64::log10)),
+            FunctionContainer::new("log1p", FunctionWrapper::Float1(f64::ln_1p)),
+            FunctionContainer::new("round", FunctionWrapper::Float1(f64::round)),
+            FunctionContainer::new("sin", FunctionWrapper::Float1(f64::sin)),
+            FunctionContainer::new("sinh", FunctionWrapper::Float1(f64::sinh)),
+            FunctionContainer::new("sqrt", FunctionWrapper::Float1(f64::sqrt)),
+            FunctionContainer::new("tan", FunctionWrapper::Float1(f64::tan)),
+            FunctionContainer::new("tanh", FunctionWrapper::Float1(f64::tanh)),
+            FunctionContainer::new("toDegrees", FunctionWrapper::Float1(f64::to_degrees)),
+            FunctionContainer::new("toRadians", FunctionWrapper::Float1(f64::to_radians)),
+            // string
+            FunctionContainer::new("concat", FunctionWrapper::Generic(concat)),
+            FunctionContainer::new("contains", FunctionWrapper::String2(contains)),
+            FunctionContainer::new("length", FunctionWrapper::String1(length)),
+            FunctionContainer::new("parseDate", FunctionWrapper::String1(parse_date)),
+            FunctionContainer::new("substring", FunctionWrapper::Generic(substring)),
+        ]
+        .iter()
+        .map(|wrap| (wrap.name, *wrap))
+        .collect()
+    };
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn unicode_length() {
+        assert_eq!(data::Value::Int(1), length("\u{2603}").unwrap());
+    }
+
+    #[test]
+    fn does_not_contain() {
+        assert_eq!(
+            data::Value::from_bool(false),
+            contains("abc", "def").unwrap()
+        );
+    }
+
+    #[test]
+    fn unicode_contains() {
```

Comment:
substring also looks fairly complex -- a quick test would be good

Reply:
Added a couple more tests and there were some substring tests in the integration test file (string_funcs_2.toml).

Related note, have you looked into a way to get test coverage?